File: src/main/java/com/ordnance/karakal/websocket/messages/PlayMessage.java
package com.ordnance.karakal.websocket.messages;

import java.util.List;
import java.util.UUID;

public class PlayMessage implements ClientMessage{
    public UUID gameId;
    public UUID playerId;

    public DRAW drawType;
    public Long drawCardId;

    public List<Long> discardIds;
}

File: frontend/src/utilities/websocket/messages/PlayMessage.ts
export type PlayMessage = {
    type: "PLAY",
    playerId: string,
    drawType: 'DECK' | 'DISCARD',
    drawCardId? : number,
    discardIds: number[]
}

export const createPlayMessage = (gameId : string, playerId: string, drawType: string, discardIds: number[], drawCardId?: number) => ({
    type: "PLAY",
    gameId,
    playerId,
    drawType,
    drawCardId,
    discardIds,
})
